### PR TITLE
Project RSpec

### PR DESCRIPTION
**Pull request by JuanPabloGil**
We test al the Enumerable methods of the project with RSpec

### CODE REVIEW

File: spec/anumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    end
+  end
+
+  describe '#my_any?' do
+    it ' elements with a length equal or bigger than three' do
+      expect(test_arr_str.my_any?(&test_length)).to be true
+    end
+
+    it 'return false when no block is given' do
+      expect(test_absolutes.my_any?).to eql(false)
```
Comment (maelfosso) [inline spec/anumerable_spec.rb]: By changing this line to 
```ruby
expect(test_absolutes.my_any?).to eql(test_absolutes.any?)
```
I got that error
```ruby
Failures:

  1) Enumerable#my_any? return false when no block is given
     Failure/Error: expect(test_absolutes.my_any?).to eql(test_absolutes.any?)
     
       expected: true
            got: false
     
       (compared using eql?)
     
       Diff:
       @@ -1,2 +1,2 @@
       -true
       +false
       
     # ./spec/anumerable_spec.rb:64:in `block (3 levels) in <top (required)>'

Finished in 0.01242 seconds (files took 0.06957 seconds to load)
21 examples, 1 failure

Failed examples:

rspec ./spec/anumerable_spec.rb:63 # Enumerable#my_any? return false when no block is given
```

File: spec/anumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  describe '#my_all?' do
+    it 'Check the length of the words 1 with 3' do
+      expect(test_arr_str.my_all?(&test_length)).to eql(true)
+    end
+
+    it 'Check the length of the words 2 with 5' do
+      expect(test_arr_str.my_all?(&test_length_2)).to eql(false)
+    end
+  end
```
Comment (maelfosso) [inline spec/anumerable_spec.rb]: By adding this test, 
```ruby
it 'return false when no block is given' do
      expect([1, false, 'hi', []].my_all?).to eql([1, false, 'hi', []].all?)
    end
```
I'm getting that error
```ruby
Failures:

  1) Enumerable#my_all? return false when no block is given
     Failure/Error: expect([1, false, 'hi', []].my_all?).to eql([1, false, 'hi', []].all?)
     
       expected: false
            got: true
     
       (compared using eql?)
     
       Diff:
       @@ -1,2 +1,2 @@
       -false
       +true
       
     # ./spec/anumerable_spec.rb:58:in `block (3 levels) in <top (required)>'
```

File: spec/anumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    end
+
+    it 'Return true when none with the test length 5' do
+      expect(test_arr_str.my_none?(test_length_2)).to be true
+    end
+
+    it 'returns true when no block given with an empty array' do
+      expect([].my_none?).to be true
+    end
+  end
```
Comment (maelfosso) [inline spec/anumerable_spec.rb]: By adding this test
```ruby
it 'return false when no block is given' do
      expect(test_absolutes.my_none?).to eql(test_absolutes.none?)
    end
```
It failed 
```ruby
Failures:

  1) Enumerable#my_none? return false when no block is given
     Failure/Error: expect(test_absolutes.my_none?).to eql(test_absolutes.none?)
     
       expected: false
            got: true
     
       (compared using eql?)
     
       Diff:
       @@ -1,2 +1,2 @@
       -false
       +true
       
     # ./spec/anumerable_spec.rb:86:in `block (3 levels) in <top (required)>'

```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changed required but you are on the right direction :smile: 

Hello @JuanPabloGil @lhscar56 

Thank you for that beautiful code :smile: I really enjoy reading it.
You really master **RSpec**

However there are 3 tests that failed after some tweaking. Please, read the comments below to know more about them.
@JuanPabloGil  I was really surprising because after approving this [PR](https://github.com/JuanPabloGil/Enumerables-Methods/pull/1) it were not supposed to failed. That means only one thing, **You change the code**. So, I invite you to check the failed test according to the the code from the [methods](https://github.com/JuanPabloGil/Enumerables-Methods/blob/methods/methods.rb), to discover what the problem is.

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: spec/anumerable_spec.rb (hunk 1 above)
Comment (JuanPabloGil) [reply spec/anumerable_spec.rb]: Solved, was a little bug with my enumerable code.

File: spec/anumerable_spec.rb (hunk 2 above)
Comment (JuanPabloGil) [reply spec/anumerable_spec.rb]: Solved, was a little bug with my enumerable code.

File: spec/anumerable_spec.rb (hunk 3 above)
Comment (JuanPabloGil) [reply spec/anumerable_spec.rb]: Solved, was a little bug with my enumerable code.

**Review by AdnanAfsari — APPROVED** [review]
**Status: Approved**
Looks good, so 👍 from me.

Happy Coding
Adnan Afsari

**Pull request merged**